Match play

Aixelsyd007 opened the pull request:
Added match play as an option for helga-jeopardy.

matthew-chandler7 committed:
first run

Aixelsyd007 committed:
added game setup; working on player join

matthew-chandler7 committed:
adding game records to reset command

Aixelsyd007 committed:
getting close

Aixelsyd007 committed:
game working

Aixelsyd007 committed:
IT WORKS!!

Aixelsyd007 committed:
more progress

Aixelsyd007 committed:
more stuff

Aixelsyd007 committed:
progress

Aixelsyd007 committed:
cleaning up

Aixelsyd007 committed:
7 second answer delay is brutal, set default to 12

will2dye4 reviewed (changes requested):
As a general note, check the spacing between functions and make sure there are two blank lines after each function before the next one.

I know you have come very far with this project already and likely don't want to undertake any major refactoring now, but I do see a couple of potential problems with the data model and think it would make more sense to reorganize it like so:

```python
{
    'game_id': 'some-uuid',
    'game_state': 'CREATED',  # or 'STARTED' or 'FINISHED'
    'host': 'will2dye4',
    'control': 'chandler',
    'players': [
        {
            'nick': 'will2dye4',
            'score': 0,
        },
        {
            'nick': 'chandler',
            'score': 0,
        },
        ...
    ],
    'categories': [
        {
            'category_id': 'cat1',
            'title': 'potent potables',
            'questions': [
                {
                    'question_id': 'clue1',
                    'question': '...',
                    'answer': '...',
                    'score': 200,
                    'active': True,
                },
                ...
            ],
        },
        ...
    ]
}
```

You could even create a `JeopardyGame` class to wrap the database queries and provide a clean API for accessing games (and fields of games) from this module, but that's probably asking too much for right now.

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

neat. I didn't know the exception had stuff in it (I'm sure there's a better way to word that)

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
added more checks to the category eval bit; added check for question_text var

Aixelsyd007 committed:
removed extra lines, replaced not joined game msgs with var; fixed indent

Aixelsyd007 committed:
more fixes per PR

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>

Aixelsyd007 committed:
Update helga_jeopardy.py

Co-authored-by: William Dye <will2dye4@mac.com>
